core/bin/external_node/src/config/mod.rs

```diff
@@ -242,6 +242,15 @@ pub struct OptionalENConfig {
     // This is intentionally not a part of `RemoteENConfig` because fetching this info from the main node would defeat
     // its purpose; the consistency checker assumes that the main node may provide false information.
     pub contracts_diamond_proxy_addr: Option<Address>,
+
+    #[serde(default = "OptionalENConfig::default_snapshots_recovery_enabled")]
+    pub snapshots_recovery_enabled: bool,
+
+    #[serde(default = "OptionalENConfig::default_pruning_chunk_size")]
+    pub pruning_chunk_size: u32,
+
+    /// If set, l1 batches will be pruned after they are that long
+    pub pruning_data_retention_hours: Option<u64>,
```

Comment:
Bikeshedding: At least for testing purposes, it may make sense to allow configuring this interval more granularly; so I'd suggest to use the seconds precision. Ideally, we could use [`humantime`](https://docs.rs/humantime-serde/latest/humantime_serde/) to parse intervals from human-readable strings, but that should be applied to all configs, and I'm not sure how well it'd work with the new config subsystem in development by @Deniallugo.

Reply:
I initially also thought about seconds, but it makes it very non-human-readable, one of the "solutions" for tests is setting it to 0 hours. We can come back to this in PR adding integration tests. I'd also rather be against introducing new library just for that.

Reply:
we can use whatever we want.

Reply:
IMO indeed hours are too coarse, also I think we don't have hours anywhere else in config. I'd do seconds and when we set value in gitops, would also add a comment like `// 5 * 60 * 60`. No need to change now.